internal/platform/platform.go

```diff
@@ -1,32 +1,82 @@
 package platform
 
 import (
-	"strings"
+	"fmt"
 
 	"github.com/jaypipes/ghw"
+	"github.com/openshift/dpu-operator/internal/daemon/plugin"
 	"sigs.k8s.io/kind/pkg/errors"
 )
 
-func supportedDpuPlatforms() []string {
-	return []string{"IPU Adapter E2100-CCQDA2"}
+type PlatformInfoProvider interface {
 }
 
-type PlatformInfoProvider interface {
-	IsDPU() (bool, error)
+type VendorDetector interface {
+	IsDpuPlatform() (bool, error)
+	VspPlugin(dpuMode bool) *plugin.GrpcPlugin
+	IsDPU(pci ghw.PCIDevice) bool
 }
 
-type PlatformInfo struct{}
+type PlatformInfo struct {
+	Detectors []VendorDetector
+}
 
-func (pi *PlatformInfo) IsDPU() (bool, error) {
-	product, err := ghw.Product()
-	if err != nil {
-		return false, errors.Errorf("Error getting product info: %v", err)
+func NewPlatformInfo() *PlatformInfo {
+	return &PlatformInfo{
+		Detectors: []VendorDetector{
+			NewIntelDetector(),
+			// add more detectors here
+		},
 	}
+}
 
-	for _, model := range supportedDpuPlatforms() {
-		if strings.Contains(product.Name, model) {
+func (pi *PlatformInfo) IsDpu() (bool, error) {
+	for _, detector := range pi.Detectors {
+		isDpu, err := detector.IsDpuPlatform()
+		if err != nil {
+			return false, err
+		}
+		if isDpu {
 			return true, nil
 		}
 	}
 	return false, nil
 }
+
+func (pi *PlatformInfo) listDpuDevices() ([]ghw.PCIDevice, []VendorDetector, error) {
+	pci, err := ghw.PCI()
+	if err != nil {
+		return nil, nil, errors.Errorf("Error getting PCI info: %v", err)
+	}
+
+	if err != nil {
+		return nil, nil, err
+	}
+	var dpuDevices []ghw.PCIDevice
+	var detectors []VendorDetector
+	for _, pci := range pci.ListDevices() {
+		for _, detector := range pi.Detectors {
+			if detector.IsDPU(*pci) {
+				dpuDevices = append(dpuDevices, *pci)
+				detectors = append(detectors, detector)
```

Comment:
We should call `detectors` something like `activeDetectors` or `currentDetectors`.

Reply:
done